### PR TITLE
Add trace details foundation

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Add a basic details view to get info on the config + trace for the selected step.

https://user-images.githubusercontent.com/1444314/112428174-fd55c700-8cf7-11eb-9977-60d66de4e3b7.mp4

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/trace/hat-trace.ts [hunk 1]
```diff
@@ -432,12 +441,53 @@ export class HaAutomationTracer extends LitElement {
...
   }
 
+  protected updated(props) {
+    super.updated(props);
+
+    // Pick first path when we load a new trace.
+    if (props.has("trace")) {
+      for (const el of Array.from(
+        this.shadowRoot!.querySelectorAll("ha-timeline")
+      )) {
```
Comment (bramkragten) [inline src/components/trace/hat-trace.ts:451]: Why not just do:
```ts
const element = this.shadowRoot!.querySelector("ha-timeline[data-path]");
if (element) {
	fireEvent(this, "value-changed", { value: element.dataset.path });
    this._selectedPath = element.dataset.path;
}
```

File: src/components/trace/hat-trace.ts [hunk 2]
```diff
@@ -432,12 +441,53 @@ export class HaAutomationTracer extends LitElement {
...
+        if (el.dataset.path) {
+          fireEvent(this, "value-changed", { value: el.dataset.path });
+          this._selectedPath = el.dataset.path;
+          break;
+        }
+      }
+    }
+
+    this.shadowRoot!.querySelectorAll("ha-timeline").forEach((el) => {
```
Comment (bramkragten) [inline src/components/trace/hat-trace.ts:460]: We are doing this for every update?

Comment (bramkragten) [inline src/components/trace/hat-trace.ts:460]:
```suggestion
    this.shadowRoot!.querySelectorAll("ha-timeline[data-path]").forEach((el) => {
```

File: src/components/trace/hat-trace.ts [hunk 3]
```diff
@@ -432,12 +441,53 @@ export class HaAutomationTracer extends LitElement {
...
+      el.style.setProperty(
+        "--timeline-ball-color",
+        this._selectedPath === el.dataset.path ? "var(--primary-color)" : null
+      );
+      if (!el.dataset.path || el.dataset.upgraded) {
```
Comment (bramkragten) [inline src/components/trace/hat-trace.ts:465]:
```suggestion
      if (el.dataset.upgraded) {
```

File: src/panels/config/automation/trace/ha-automation-trace.ts [hunk 4]
```diff
@@ -254,6 +285,16 @@ export class HaAutomationTrace extends LitElement {
           top: 8px;
           right: 8px;
         }
+
+        .details {
+          display: flex;
```
Comment (bramkragten) [inline src/panels/config/automation/trace/ha-automation-trace.ts:290]: So it is not sticked to the sides
```suggestion
          display: flex;
          margin: 0 16px;
```

File: src/components/trace/hat-trace.ts (hunk 2 above)
Comment (balloob) [reply src/components/trace/hat-trace.ts:460]: Yeah but I guard if it's upgraded.

Comment (bramkragten) [reply src/components/trace/hat-trace.ts:460]: We still query and run over it? Can't we just only do it on change of `trace`?

Comment (balloob) [reply src/components/trace/hat-trace.ts:467]: actually, we can't. We still need to set the "active" color.

Comment (bramkragten) [reply src/components/trace/hat-trace.ts]: that is `_selectedPath` right?

**Pull request merged**